Added code changes for inbound endpoint and outbound endpoint

arpit-gagneja opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

Recently, we have changed our swagger for a couple of resources.  So that necessitated generating new cmdlets for changed resources. Our checked-in public swagger is here: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/dnsresolver/resource-manager/Microsoft.Network/preview/2020-04-01-preview/dnsresolver.json.

In the past few releases, DNSResolver was a special module whose code was checked into the generation branch but not moved to the main branch (so it was never publicly released). Now, it will be checked into the main branch as well as public swagger has already been checked in and the SDK is ready for the public preview.

To be mentioned in Release Notes:

Az.DnsResolver
- Initial public review release
- The below cmdlets are added to manage top-level resource which has to be created for both inbound scenario and outbound scenario. The inbound scenario lets customers help DNS queries from on-premise networks to be resolved from Azure private DNS zones. And the outbound scenario lets customers help DNS queries from Azure to be resolved from the customer configured target DNS server.
	- New-AzDnsResolver
	- Get-AzDnsResolver
	- Remove-AzDnsResolver
- The below cmdlets are added to manage resources for the inbound scenario.
	- New-AzDnsResolverInboundEndpoint
	- Get-AzDnsResolverInboundEndpoint
	- Remove-AzDnsResolverInboundEndpoint
- The below cmdlets are added to manage resources for the outbound scenario.
	- New-AzDnsResolverOutboundEndpoint
	- New-AzDnsForwardingRuleset
	- New-AzDnsForwardingRulesetVirtualNetworkLink
	- New-AzDnsForwardingRulesetForwardingRule
	- Get-AzDnsResolverOutboundEndpoint
	- Get-AzDnsForwardingRuleset
	- Get-AzDnsForwardingRulesetVirtualNetworkLink
	- Get-AzDnsForwardingRulesetForwardingRule
	- Remove-AzDnsResolverOutboundEndpoint
	- Remove-AzDnsForwardingRuleset
	- Remove-AzDnsForwardingRulesetVirtualNetworkLink
	- Remove-AzDnsForwardingRulesetForwardingRule

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

isra-fel commented:
@VeryEarly Hey Yabo, some background info - in the past few releases DNSResolver was a special module whose code was checked in to generation branch but not moved to main branch (so it was never publicly released).
Now the service team has checked in swagger in public repo and is ready to release Az.DnsResolver for the first time.

isra-fel commented on src/DnsResolver/README.md:
```diff
@@ -75,7 +75,7 @@ require:
 input-file:
   - $(this-folder)/resources/specification/dnsresolver/resource-manager/Microsoft.Network/preview/2020-04-01-preview/dnsresolver.json
```
Now that swagger is published, there's no need to use a local swagger file to generate anymore. Here's an example of how to configure it in readme
```yaml
branch: 7a2cc29033fe4027ef421267f1684efbd0d40a93
input-file:
  - $(repo)/specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json
```

arpit-gagneja replied:
Changed as discussed.

muwaqar reviewed (dismissed):
Please update Swagger to the official public one as posted by isra-fel

arpit-gagneja commented:
> Please update Swagger to the official public one as posted by isra-fel

I have updated as discussed.

arpit-gagneja commented:
@VeryEarly, can you please take a quick look? We want to check in changes
into the main branch this week so that we can meet this month’s release
cut-off date.

On Fri, Feb 11, 2022 at 6:38 AM Muhammad Waqar ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Please update Swagger to the official public one as posted by isra-fel
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/azure-powershell/pull/17083#pullrequestreview-880262719>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEUUXZ6PKDZTRG2H472AGITU2UNN3ANCNFSM5ODRYM2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

VeryEarly commented on src/DnsResolver/README.md:
```diff
@@ -73,9 +73,9 @@ In this directory, run AutoRest:
 require:
```
please specify the latest commit of swagger spec as branch:
```
branch: ${commit}
```

arpit-gagneja replied:
Added the commit.

VeryEarly commented:
Please delete swagger spec in this folder

arpit-gagneja commented:
> Please delete swagger spec in this folder

Removed the unneeded files.

arpit-gagneja commented:
@VeryEarly, can you please take a look again? I have addressed the comments.

On Mon, Feb 14, 2022 at 12:05 AM Yabo Hu ***@***.***> wrote:

> Please delete swagger spec in this folder
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/azure-powershell/pull/17083#issuecomment-1038771549>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEUUXZ52M3EIW5DZDV26O33U3CZVHANCNFSM5ODRYM2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
